sstables/mx/reader.cc

```diff
@@ -559,6 +559,14 @@ class mp_row_consumer_m {
             if (!_cells.empty()) {
                 fill_cells(column_kind::regular_column, _in_progress_row->cells());
             }
+            if (_slice.options.contains(query::partition_slice::option::reversed) &&
+                // we always consume whole rows (i.e. `consume_row_end` is always called) when reading in reverse,
+                // even when `consume_row_start` requested to ignore the row (TODO: explain why this happens).
```

Comment:
Why not just implement skipping?

Reply:
It would require translating the mx reader offsets to the data file offsets (they are different due to the "partition reversing data source" lying in between), which apparently is complicated. I'm not sure I fully understand the details too. @wmitros, the original author of the reversing data source explained:

> It should be possible to translate the data positions of the
index_reader to the positions of the intermediary data stream,
but I'm not sure we want to do that, as it add some complexity
in the index_reader and the intermediary stream. One reason
for the complexity in the input streamis that we would have two
kinds of skips: a single row skip, and a skip to a clustering
range. We know the offset of the next row, so we could check that
to differentiate them. We would also need to add an information
about the position of first clustering row and end of the last one
in the index_reader.

We do have skipping, but it's implemented differently: we advance the index and the source notices it.

BTW. I forgot about this TODO, need to update the comment in next version.